docs/investigating_a_regression.md

```diff
@@ -0,0 +1,192 @@
+OUT-07-21:MLUPS: 587.6702557143142
+OUT-07-25:MLUPS: 577.3063618431178
+OUT-07-26:MLUPS: 577.2362978080912
+OUT-07-27:MLUPS: 577.2101850145785
+OUT-07-28:MLUPS: 577.0716349809895
+OUT-07-29:MLUPS: 577.4223280707176
+OUT-07-30:MLUPS: 577.2255967221336
+OUT-08-01:MLUPS: 577.277685388252
+OUT-08-02:MLUPS: 577.0137874289354
+OUT-08-03:MLUPS: 577.1333281553946
+OUT-08-04:MLUPS: 577.305012020407
+OUT-08-05:MLUPS: 577.2143988866626
+OUT-08-06:MLUPS: 577.2409145495443
+OUT-08-07:MLUPS: 577.2602819927345
+OUT-08-08:MLUPS: 577.2823738293221
+OUT-08-09:MLUPS: 577.3453199728248
+OUT-08-11:MLUPS: 577.3161423260563
+OUT-08-12:MLUPS: 577.1697775786824
+OUT-08-13:MLUPS: 577.3049883393633
+OUT-08-14:MLUPS: 576.9051978525331
+OUT-08-15:MLUPS: 577.5331743016213
+OUT-08-16:MLUPS: 577.5117505070573
+OUT-08-18:MLUPS: 577.5930698237612
+OUT-08-19:MLUPS: 577.3539885757353
+OUT-08-20:MLUPS: 577.4190113959127
+OUT-08-21:MLUPS: 577.300394253605
+OUT-08-22:MLUPS: 577.4263792037783
+OUT-08-23:MLUPS: 577.4087536357031
+OUT-08-24:MLUPS: 577.1094728438082
+OUT-08-25:  File "/XLB/examples/performance/MLUPS3d.py", line 5, in <module>
+OUT-08-26:MLUPS: 537.0164618489928
+OUT-08-27:MLUPS: 536.9545448661609
+OUT-08-28:MLUPS: 536.2887650464874
+OUT-08-29:MLUPS: 536.7178471720636
+OUT-08-30:MLUPS: 536.6978912984252
+OUT-09-01:MLUPS: 536.7030899164106
+OUT-09-04:MLUPS: 536.5339818238837
+OUT-09-05:MLUPS: 536.6507808565617
+OUT-09-06:MLUPS: 536.7144494518315
+OUT-09-08:MLUPS: 536.7376612408998
+OUT-09-09:MLUPS: 536.7798324141778
+OUT-09-10:MLUPS: 536.726157440174
+OUT-09-11:MLUPS: 536.7446210750584
+OUT-09-12:MLUPS: 536.6707332269023
+OUT-09-13:MLUPS: 536.6777936517823
+OUT-09-14:MLUPS: 536.7581523280307
+OUT-09-15:MLUPS: 536.6156273667873
+OUT-09-16:MLUPS: 536.7320935035265
+OUT-09-17:MLUPS: 536.7104991444398
+OUT-09-18:MLUPS: 536.7492269469092
+OUT-09-19:MLUPS: 536.6760131792959
+OUT-09-20:MLUPS: 536.7361260076634
+```
+
+This found that 8-24 was good, but 8-26 was bad. On 8-25 there was
+another issue that prevented from getting results. So we need to
+investigate hourly between 8-24 and 8-26. There was a smaller slowdown
+earlier, lets ignore it for this example. It would be only another
+hourly investigation between those dates.
+
+## Hourly investigation.
+
+This does a checkout of JAX and XLA at each hour between the 2 dates,
+rebuilds everything and runs the test.  The scripts are structured
+differently. We start the working container and keep it. Then inside
+it, we only trigger incremental XLA builds except for the first
+build. So it is much faster after the first iteration.
+
+- test_runner2.sh:
+```
+  # Execute this script inside the container:
+  # docker run -v $PWD:/dir --gpus=all ghcr.io/nvidia/jax:nightly-2023-08-24 /bin/bash
+  cd /opt/xla-source
+  git remote update
+  cd /opt/jax-source
+  git remote update
+  pip install jmp pyvista numpy matplotlib Rtree trimesh jmp termcolor orbax
+  cd /tmp
+  git clone https://github.com/Autodesk/XLB
+  cd XLB
```

Comment:
These two lines don't seem to belong here?

Reply:
Here, we are doing a linear search on the nightly container.
They don't have the software we want to test nor its dependencies installed above.
The clone provide used by this line in the file test2.sh:
```python3 examples/performance/MLUPS3d.py 256 200```

I don't have an idea where to place them otherwise. If you have, please tell me.